nomad/rpc.go

```diff
@@ -99,15 +100,23 @@ func (r *rpcHandler) listen(ctx context.Context) {
 				return
 			}
 
-			select {
-			case <-ctx.Done():
-				return
-			default:
+			if tempDelay == 0 {
```

Comment:
I think this code is just complex enough to warrant it's own `r.handleAcceptErr(err)` func. It looks to me we're safe to stick the tempDelay into the struct (although I think it should be renamed to be more descriptive), and do the sleep in the new func.

Reply:
@nickethier , the return on `r.shutdown` and the continue (if we keep it) have to stay inside if the `if err` block, so only the sleep logic can go into `handleAcceptErr`. however, the `r.acceptLoopDelay` has to stay outside of the if block to reset the counter if `err == nil`, and i would be worried about the logic for `acceptLoopDelay` living in two places.

for example: 
https://gist.github.com/cgbaker/ab90f40bcbdd97c28c057e768f39bfc9